src/Core/BackgroundSchedulePool.cpp

```diff
@@ -102,9 +108,17 @@ void BackgroundSchedulePoolTaskInfo::execute()
         executing = true;
     }
 
+    /// Using this tmp query_id storage to prevent bad_alloc thrown under the try/catch.
+    std::string task_query_id = fmt::format("{}::{}", pool.thread_name, UUIDHelpers::generateV4());
+
     try
     {
+        chassert(current_thread); /// Thread from global thread pool
+        current_thread->setQueryId(std::move(task_query_id));
+
         function();
+
+        task_query_id = current_thread->clearQueryId();
```

Comment:
Do we need this?
Can we just use `query_id.clear()` in `clearQueryId` and return `void`?

Reply:
The original idea was to not use allocator at all, but it seems deallocate should be noexcept.